torch/utils/data/dataloader.py

```diff
@@ -48,8 +57,40 @@ def _worker_loop(dataset, index_queue, data_queue, collate_fn, seed, init_fn, wo
     if init_fn is not None:
         init_fn(worker_id)
 
+    manager_handle = None
+    if IS_WINDOWS:
+        kernel32 = ctypes.WinDLL('kernel32', use_last_error=True)
+        kernel32.OpenProcess.argtypes = (DWORD, BOOL, DWORD)
+        kernel32.OpenProcess.restype = HANDLE
+        kernel32.WaitForSingleObject.argtypes = (HANDLE, DWORD)
+        kernel32.WaitForSingleObject.restype = DWORD
+
+        # Value obtained from https://msdn.microsoft.com/en-us/library/ms684880.aspx
+        SYNCHRONIZE = 0x00100000
+        manager_handle = kernel32.OpenProcess(SYNCHRONIZE, 0, manager_pid)
+
+        if not manager_handle:
+            raise ctypes.WinError(ctypes.get_last_error())
+
+    def is_manager_process_alive(manager_pid, manager_handle):
+        if IS_WINDOWS:
+            result = kernel32.WaitForSingleObject(manager_handle, 0)
+            # Value obtained from https://msdn.microsoft.com/en-us/library/windows/desktop/ms687032.aspx
+            if result == 0:
+                return False
+        else:
+            if os.getppid() != manager_pid:
+                return False
+        return True
```

Comment:
nit: this could easily be collapsed to
```python
if IS_WINDOWS:
    return kernel23.WaitForSingleObject(...) == 0
else:
    return os.getppid() == manager_pid
```